crates/fuel-core/src/service/adapters/txpool.rs

```diff
@@ -126,6 +181,25 @@ impl fuel_core_txpool::ports::PeerToPeer for P2PAdapter {
     ) -> anyhow::Result<()> {
         Ok(())
     }
+
+    fn new_tx_subscription(&self) -> BoxStream<Vec<u8>> {
+        Box::pin(fuel_core_services::stream::pending())
+    }
+
+    async fn request_tx_ids(
+        &self,
+        _peer_id: Vec<u8>,
+    ) -> anyhow::Result<Vec<fuel_core_txpool::types::TxId>> {
+        Ok(vec![])
```

Comment:
if we wrapped the vec in an option, we could just do `Ok(None)`. perhaps this is better for performance, but this will never be called if p2p isn't enabled, so it doesn't matter i guess

Reply:
This doesn't change much as it's never called and will bring a lot of overhead everywhere in the code to manage the option

Reply:
agreed!